logdna/adapter/adapter.go

```diff
@@ -141,24 +141,24 @@ func (adapter *Adapter) Stream(logstream chan *router.Message) {
 // readQueue is a method for reading from queue:
 func (adapter *Adapter) readQueue() {
 
-    buffer := make([]Line, 0, adapter.Config.MaxBufferSize)
+    buffer := make([]Line, 0)
     timeout := time.NewTimer(adapter.Config.FlushInterval)
 
     for {
         select {
         case msg := <-adapter.Queue:
-            if len(buffer) >= cap(buffer) {
+            if binary.Size(buffer) >= int(adapter.Config.MaxBufferSize) {
```

Comment:
I'd advise keeping track of the size of the buffer in a var like `bufferSize`.
`binary.Size(buffer)` will serialize the buffer each iteration to get the size of the buffer, that's quite expensive.

Reply:
Agree considering there is another comparison involving `len` - how does it look now?